### DISCUSSION TITLE
Auth Callout with leaks

### DISCUSSION THREAD

**Discussion by mrosm20** [nats-io/nats-server#5813] [Q&A]
Is it possible to use an auth-callout configured through NSC  in a leaf-spoke topology. The leaf remote uses the  Auth account and auth.creds which is the same account for the Auth callout. Everything is connect and working when the system comes up. However, when  i  connect the client  to the leaf i get a   timeout error. When  i connect the client to the  hub the auth call-out is called properly. Help is much appreciated.

**Comment by aricart** [comment]
The operator on the leaf is different from the one on the hub, yes?

**Reply by aricart** [reply]
After thinking about it. The leaf should be in its own authentication domain, so Auth for the leaf should be in a call out on the leaf.

**Reply by mrosm20** [reply]
the first test the operator is the same.  can it be done? or does all leafs and hub be direct connected?

**Reply by aricart** [reply]
Operator for the leaf must be different

**Comment by aricart** [comment]
The callout on the leaf should be authenticating within the leaf. The leaf connection will bridge the local traffic on the hub if the user maps to an account that is bridged through the leaf.

If you are trying to authenticate by generating credentials on the hub, you should be clustering, not leafing.

**Reply by mrosm20** [reply]
i think that would be fine except can you have different jetstream rules on clustered nodes.
